app/vminsert/influx/request_handler.go

```diff
@@ -105,9 +106,15 @@ func insertRows(at *auth.Token, db string, rows []parser.Row, mayOverrideAccount
 			ic.Labels = ic.Labels[:len(ic.Labels)-1]
 			ic.AddLabel("", metricGroup)
 			ic.MetricNameBuf = storage.MarshalMetricLabelRaw(ic.MetricNameBuf[:metricNameBufLen], placeholderLabel)
-			storageNodeIdx := ic.GetStorageNodeIdx(&atCopy, ic.Labels)
-			if err := ic.WriteDataPointExt(&atCopy, storageNodeIdx, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
-				return err
+			storageNodeGroupIDs := ic.GetStorageNodeGroupIds(&atCopy, ic.Labels)
+			for _, storageNodeGroupID := range storageNodeGroupIDs {
+				if err := ic.WriteDataPointExt(at, storageNodeGroupID, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
```

Comment:
It is better passing a pointer to replication group instead of its name to `WriteDataPointExt`. Something like the following:

```go
ic.WriteDataPointExt(at, rg, storageNodeIdx, ic.MetricNameBuf, r.Timestamp, f.Value)
```

This will eliminate CPU overhead on replication group lookup - `replicationGroups[groupName]` inside `WriteDataPointExt`